mkosi/architecture.py

```diff
@@ -0,0 +1,133 @@
+            "armv5tb"     : Architecture.arm,
+            "armv4tl"     : Architecture.arm,
+            "armv4tb"     : Architecture.arm,
+            "armv4l"      : Architecture.arm,
+            "armv4b"      : Architecture.arm,
+            "alpha"       : Architecture.alpha,
+            "arc"         : Architecture.arc,
+            "arceb"       : Architecture.arc,
+            "x86_64"      : Architecture.x86_64,
+            "i686"        : Architecture.x86,
+            "i586"        : Architecture.x86,
+            "i486"        : Architecture.x86,
+            "i386"        : Architecture.x86,
+            "ia64"        : Architecture.ia64,
+            "parisc64"    : Architecture.parisc,
+            "parisc"      : Architecture.parisc,
+            "loongarch64" : Architecture.loongarch64,
+            "mips64"      : Architecture.mips64_le,
+            "mips"        : Architecture.mips_le,
+            "ppc64le"     : Architecture.ppc64_le,
+            "ppc64"       : Architecture.ppc64,
+            "ppc"         : Architecture.ppc,
+            "riscv64"     : Architecture.riscv64,
+            "riscv32"     : Architecture.riscv32,
+            "riscv"       : Architecture.riscv64,
+            "s390x"       : Architecture.s390x,
+            "s390"        : Architecture.s390,
+            "tilegx"      : Architecture.tilegx,
+        }.get(s)
+
+        if not a:
+            die(f"Architecture {a} is not supported")
+
+        return a
+
+    def to_discoverable_partitions_spec(self) -> str:
+        return str(self)
+
+    def to_efi(self) -> str:
+        a = {
+            Architecture.x86_64      : "x64",
+            Architecture.x86         : "ia32",
+            Architecture.arm64       : "aa64",
+            Architecture.arm         : "arm",
+            Architecture.riscv64     : "riscv64",
+            Architecture.loongarch64 : "loongarch64",
+        }.get(self)
+
+        if not a:
+            die(f"Architecture {self} does not support UEFI")
+
+        return a
+
+    def to_qemu(self) -> str:
+        a = {
+            Architecture.alpha: "alpha",
+            Architecture.arm: "arm",
+            Architecture.arm64: "aarch64",
+            Architecture.loongarch64: "loongarch64",
+            Architecture.mips64_le: "mips",
+            Architecture.mips_le: "mips",
+            Architecture.parisc: "hppa",
+            Architecture.ppc: "ppc",
+            Architecture.ppc64: "ppc",
+            Architecture.ppc64_le: "ppc",
+            Architecture.riscv32: "riscv32",
+            Architecture.riscv64: "riscv64",
+            Architecture.s390x: "s390x",
+            Architecture.x86: "i386",
+            Architecture.x86_64: "x86_64",
+        }.get(self)
+
+        if not a:
+            die(f"Architecture {self} not supported by QEMU")
+
+        return a
+
+    def is_native(self) -> bool:
+        return self == native_architecture()
+
+
+def native_architecture() -> Architecture:
```

Comment:
As a alternative constructor I think that should live in `Architecture`.
```suggestion
@ classmethod
def native_architecture(cls) -> "Architecture":
    return cls.from_uname(platform.machine())
```